dj_cqrs/_validation.py

```diff
@@ -0,0 +1,165 @@
+            # TODO: raise error in 2.0.0
+            logger.warning(
+                'Settings CQRS_MESSAGE_TTL=%s is invalid, using default %s.',
+                message_ttl, DEFAULT_MASTER_MESSAGE_TTL,
+            )
+            master_settings['CQRS_MESSAGE_TTL'] = DEFAULT_MASTER_MESSAGE_TTL
+    else:
+        master_settings['CQRS_MESSAGE_TTL'] = DEFAULT_MASTER_MESSAGE_TTL
+
+    correlation_func = master_settings.get('correlation_function')
+    if not correlation_func:
+        master_settings['correlation_function'] = None
+    elif not callable(correlation_func):
+        raise AssertionError('CQRS master correlation_function must be callable.')
+
+
+def _validate_replica(cqrs_settings):
+    queue = cqrs_settings.get('queue')
+    assert queue, 'CQRS queue is not set.'
+    assert isinstance(queue, str), 'CQRS queue must be string.'
+
+    default_replica_settings = {
+        'replica': {
+            'CQRS_MAX_RETRIES': DEFAULT_REPLICA_MAX_RETRIES,
+            'CQRS_RETRY_DELAY': DEFAULT_REPLICA_RETRY_DELAY,
+            'CQRS_DELAY_QUEUE_MAX_SIZE': DEFAULT_REPLICA_DELAY_QUEUE_MAX_SIZE,
+        },
+    }
+
+    if 'replica' not in cqrs_settings:
+        cqrs_settings.update(default_replica_settings)
+        return
+
+    replica_settings = cqrs_settings['replica']
+    assert isinstance(replica_settings, dict), 'CQRS replica configuration must be dict.'
+
+    if 'CQRS_MAX_RETRIES' in replica_settings:
+        min_retries = 0
+        max_retries = replica_settings['CQRS_MAX_RETRIES']
+        if (max_retries is not None) and (
+            not isinstance(max_retries, int) or max_retries < min_retries
+        ):
+            # No error is raised for backward compatibility
+            # TODO: raise error in 2.0.0
+            logger.warning(
+                'Replica setting CQRS_MAX_RETRIES=%s is invalid, using default %s.',
+                max_retries, DEFAULT_REPLICA_MAX_RETRIES,
+            )
+            replica_settings['CQRS_MAX_RETRIES'] = DEFAULT_REPLICA_MAX_RETRIES
+    else:
+        replica_settings['CQRS_MAX_RETRIES'] = DEFAULT_REPLICA_MAX_RETRIES
+
+    min_retry_delay = 0
+    retry_delay = replica_settings.get('CQRS_RETRY_DELAY')
+    if 'CQRS_RETRY_DELAY' not in replica_settings:
+        replica_settings['CQRS_RETRY_DELAY'] = DEFAULT_REPLICA_RETRY_DELAY
+    elif not isinstance(retry_delay, int) or retry_delay < min_retry_delay:
+        # No error is raised for backward compatibility
+        # TODO: raise error in 2.0.0
+        logger.warning(
+            'Replica setting CQRS_RETRY_DELAY=%s is invalid, using default %s.',
+            retry_delay, DEFAULT_REPLICA_RETRY_DELAY,
+        )
+        replica_settings['CQRS_RETRY_DELAY'] = DEFAULT_REPLICA_RETRY_DELAY
+
+    # delay_queue_max_size - is the wrong name for CQRS_DELAY_QUEUE_MAX_SIZE,
+    #  which is supported for backward compatibility
+    # TODO: drop `delay_queue_max_size` in 2.0.0
+    min_delay_size = 0
```

Comment:
delay_size sounds like some Time measure (seconds, etc.), not Queue size.
For me it's better to stick close to settings naming (DELAY_QUEUE_MAX_SIZE).